src/actions/assetsActions.js

```diff
@@ -630,14 +642,43 @@ export const resetSearchAssetsResultAction = () => ({
   type: RESET_ASSETS_SEARCH_RESULT,
 });
 
-export const checkForMissedAssetsAction = (transactionNotifications: Object[]) => {
+const getSupportedTokensAction = (supportedAssets: Asset[], currentAssets: AssetsByAccount, account: Account) => {
+  const accountId = getAccountId(account);
+  const currentAccountAssets = get(currentAssets, accountId, {});
+  const currentAccountAssetsTickers = Object.keys(currentAccountAssets);
+
+  // HACK: Dirty fix for users who removed somehow ETH and PLR from their assets list
+  if (!currentAccountAssetsTickers.includes(ETH)) currentAccountAssetsTickers.push(ETH);
+  if (!currentAccountAssetsTickers.includes(PLR)) currentAccountAssetsTickers.push(PLR);
+
+  const updatedAccountAssets = supportedAssets
+    .filter(asset => currentAccountAssetsTickers.includes(asset.symbol))
+    .reduce((memo, asset) => ({ ...memo, [asset.symbol]: asset }), {});
+  return { id: accountId, ...updatedAccountAssets };
+};
+
+const getAllOwnedAssetsAction = (accountId: string, supportedAssets) => {
+  return async (dispatch: Dispatch, getState: GetState, api: Object) => {
+    const addressErc20Tokens = await api.getAddressErc20TokensInfo(accountId); // all address' assets except ETH;
+    const accOwnedErc20Assets = {};
+    if (addressErc20Tokens.length) {
+      addressErc20Tokens.forEach((token) => {
+        const tokenTicker = get(token, 'tokenInfo.symbol', '');
+        const supportedAsset = supportedAssets.find(asset => asset.symbol === tokenTicker);
+        if (supportedAsset && !accOwnedErc20Assets[tokenTicker]) accOwnedErc20Assets[tokenTicker] = supportedAsset;
+      });
+    }
+    return accOwnedErc20Assets;
```

Comment:
Just a question; Isn't it con to use the return of a `dispatch()` call? I feel like `dispatch()` calls should be "fire and forget". In this particular case, it doesn't seem like we need either `dispatch` or `getState`, so we don't even need to wrap this on a `dispatch()` call just to get the `api` object . What if instead of having

```js
const getAllOwnedAssetsAction = (...) => {
  return async (..., api: Object) {
    // ...
  };
};
```

We have just:

```js
const getAllOwnedAssetsAction = async (..., api: Object) => {
  // ...
};
```
..? The caller should pass the `api` object, but this function is gonna be a bit less obscure, and the caller doesn't have to wait for a `dispatch()`. Thoughts?

Reply:
In some exceptional situations, we could return something. But normally we should update the state and the `getAllOwnedAssetsAction()` caller should await for the method to be completed and get the updated state

Reply:
```await dispatch(getAllOwnedAssetsAction());
const smth  = get(getState(), 'whatever');
```

Reply:
Hmmm.. Yeah, I see your point, but I feel like adding an `await` to the `dispatch()` call goes against the redux design to be asynchronous. I understand exceptions can exist, but then exceptions should be there only when there's no other feasible way around it.

Not sure of other cases, but on this one it looks like this function doesn't even need to go through the dispatch queue.

Reply:
redux design to be asynchronous slightly conflicts with the real-life use-cases :D that's why someone created a redux-saga project ;)

regarding your suggestion - yes, we can remove the `return async (..., api: Object) {` and in the caller instead of writing `dispatch(getAllOwnedAssetsAction())` we would call as `dispatch(getAllOwnedAssetsAction)`. As for me, it looks nearly the same as it was before, although we'll have 99% of code written in one style plus that method written in slightly another style. I think that moment would create even more confusion for the future devs

Reply:
> instead of writing dispatch(getAllOwnedAssetsAction()) we would call as dispatch(getAllOwnedAssetsAction)

My point is that we don't need the `dispatch` there. And it doesn't look like an action either, so we could remove the `Action` part from the name, even move it to another file if there's confusion. It would be just `const assets = await getAllOwnedAssets(accountId, supportedAssets, api)`.

Reply:
Just curious, how would you extract the api variable out of the redux middleware?

Reply:
sorry, just found that we're dispatching it not from the component, but from another action

Reply:
yes, then I agree, we can move it out from actions into services, remove the dispatch and pass the `api` as a param. Not a critical thing, mostly nice-to-have

Reply:
and then we could safely use the return statement, without breaking the redux `dispatch only` pattern

Reply:
Indeed, not a critical thing :+1:

Reply:
Removed dispatch, but I've left this helper method in the same file - we are not reusing anywhere else